Kuroba/app/src/main/java/com/github/adamantcheese/chan/features/gesture_editor/EditableZone.kt

```diff
@@ -0,0 +1,284 @@
+package com.github.adamantcheese.chan.features.gesture_editor
+
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.PointF
+import com.github.adamantcheese.chan.utils.AndroidUtils.dp
+
+class EditableZone {
+    private var currentAttachSide = AttachSide.Left
+    private var viewWidth = 0f
+    private var viewHeight = 0f
+
+    // This represent whether the whole zone is being moved by the user
+    private var isMoving = false
+
+    // This represent whether the zone is being resized by the user
+    private var isResizing = false
+    private var zoneOriginToTouchPosDeltaWhenMoving = PointF(0f, 0f)
+
+    private val zoneDefaultWidth = dp(32f).toFloat()
+    private val zoneDefaultHeight = dp(32f).toFloat()
+    private val handleSize = dp(16f).toFloat()
+    private val zone = SafeRectF(0f, 0f, MINIMUM_SIZE, MINIMUM_SIZE)
```

Comment:
I think this is not correct. It should be SafeRectF(0f, 0f, MINIMUM_SIZE, MINIMUM_SIZE, MINIMUM_SIZE, MAXIMUM_SIZE)